client/ayon_nuke/plugins/publish/validate_asset_context.py

```diff
@@ -110,5 +112,29 @@ def repair(cls, instance):
         )
         for _key in invalid_keys:
             created_instance[_key] = instance.context.data[_key]
-
         create_context.save_changes()
+        if instance.data["productType"] in {"prerender", "render", "image"}:
+            cls.reset_write_node_filepath(instance)
+
+    @classmethod
+    def reset_write_node_filepath(cls, instance):
+        instance_node = instance.data["transientData"]["node"]
+        write_node = nuke.allNodes(group=instance_node, filter="Write")[0]
+        data = dict({
+            "fpath_template": (
+            "{work}/renders/nuke/{subset}/{subset}.{frame}.{ext}"),
+            "ext": write_node["file_type"].value(),
+            "folderPath": instance.context.data["folderPath"],
+            "task": instance.context.data["task"],
+            "productName": instance.data["productName"],
+            "productType": instance.data["productType"]
+        })
+        anatomy_filled = format_anatomy(data)
+
+        # build file path to workfiles
+        fdir = str(
+            anatomy_filled["work"]["default"]["directory"]
+        ).replace("\\", "/")
+        data["work"] = fdir
+        fpath = StringTemplate(data["fpath_template"]).format_strict(data)
+        write_node["file"].setValue(fpath)
```

Comment:
Shouldn't this be up to the Creator class instead to update the filepath if the context changes?

Fusion does that for example:
- https://github.com/ynput/ayon-fusion/blob/develop/client/ayon_fusion/api/plugin.py#L121-L138

That way it also works if you change the context via the publisher UI - not just in the validator. (And you can remove this logic from this validator then because it's already updating via the Creator above)

Reply:
I added a separate function for updating instances and it works for saving changes but it doesn't work in repair action. It is mostly because we need to access the transient data to get the filepath and fix it? 
The repair action still needs `reset_write_nodes_filepath` to fix the filepath.